BGL/include/CGAL/boost/graph/Connected_component_graph.h

```diff
@@ -0,0 +1,585 @@
+
+    //act as a border
+    typename boost::graph_traits< Connected_component_graph<Graph, FaceComponentMap> >::halfedge_descriptor hcirc = h;
+    do
+    {
+        if(is_valid(hcirc, w))
+        {
+            return hcirc;
+        }
+        hcirc = opposite(next(hcirc,w.graph()),w.graph());
+    }while(hcirc != h);
+    return boost::graph_traits< CGAL::Connected_component_graph<Graph, FaceComponentMap> >::null_halfedge();
+}
+
+template <class Graph, typename FaceComponentMap >
+typename boost::graph_traits< Connected_component_graph<Graph, FaceComponentMap> >::halfedge_descriptor
+prev(typename boost::graph_traits< Connected_component_graph<Graph, FaceComponentMap> >::halfedge_descriptor h,
+     const Connected_component_graph<Graph, FaceComponentMap> & w)
+{
+
+    CGAL_assertion(is_valid(h, w));
+    if(is_valid(face(h, w.graph()), w))
+        return prev(h, w.graph());
+
+    //act as a border
+    typename boost::graph_traits< Connected_component_graph<Graph, FaceComponentMap> >::halfedge_descriptor hcirc = h;
+    do
+    {
+        if(is_valid(hcirc, w))
+        {
+            return hcirc;
+        }
+        hcirc = opposite(prev(hcirc,w.graph()), w.graph());
+    }while(hcirc != h);
+    return boost::graph_traits< CGAL::Connected_component_graph<Graph, FaceComponentMap> >::null_halfedge();
+}
+
+//
+// HalfedgeListGraph
+//
+
+template <class Graph, typename FaceComponentMap >
+std::pair<typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::halfedge_iterator,
+typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::halfedge_iterator>
+halfedges(const Connected_component_graph<Graph, FaceComponentMap> & w)
+{
+    typedef typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::halfedge_iterator halfedge_iterator;
+    typedef typename boost::graph_traits<Graph >::halfedge_iterator g_halfedge_iterator;
+
+    typename Connected_component_graph<Graph, FaceComponentMap> ::Is_simplex_valid predicate(&w);
+    std::pair<g_halfedge_iterator, g_halfedge_iterator> original_halfedges = halfedges(w.graph());
+
+    return make_range(halfedge_iterator(predicate, original_halfedges.first, original_halfedges.second),
+                      halfedge_iterator(predicate, original_halfedges.second, original_halfedges.second));
+}
+
+
+template <class Graph, typename FaceComponentMap >
+typename boost::graph_traits<Graph>::halfedges_size_type
+num_halfedges(const Connected_component_graph<Graph, FaceComponentMap> & w)
+{
+    return num_halfedges(w.graph());
+}
+
+// FaceGraph
+template <class Graph, typename FaceComponentMap >
+typename boost::graph_traits< Connected_component_graph<Graph, FaceComponentMap> >::face_descriptor
+face(typename boost::graph_traits< Connected_component_graph<Graph, FaceComponentMap> >::halfedge_descriptor h,
+     const Connected_component_graph<Graph, FaceComponentMap> & w)
+{
+    CGAL_assertion(CGAL::is_valid(h, w));
```

Comment:
if `h` is a border halfedge (on the border of the patch pointing outside the patch), `null_face()` should be returned.